pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TsvParticipantExporter.java

```diff
@@ -20,28 +20,19 @@ public TsvParticipantExporter(List<ModuleExportConfig> moduleConfigs,
     }
 
 
-    public void export(Response response) throws IOException {
-        setResponseHeaders(response);
+    public void export(OutputStream os) throws IOException {
+        PrintWriter printWriter = new PrintWriter(os);
+        List<String> headerRowValues = getHeaderRow();
+        List<String> subHeaderRowValues = getSubHeaderRow();
 
-        try (PrintWriter writer = response.raw().getWriter()) {
-            List<String> headerRowValues = getHeaderRow();
-            List<String> subHeaderRowValues = getSubHeaderRow();
-
-            writer.println(getRowString(headerRowValues));
-            writer.println(getRowString(subHeaderRowValues));
-            for (Map<String, String> valueMap : participantValueMaps) {
-                List<String> rowValues = getRowValues(valueMap, headerRowValues);
-                String rowString = getRowString(rowValues);
-                writer.println(rowString);
-            }
-            writer.flush();
+        printWriter.println(getRowString(headerRowValues));
+        printWriter.println(getRowString(subHeaderRowValues));
+        for (Map<String, String> valueMap : participantValueMaps) {
+            List<String> rowValues = getRowValues(valueMap, headerRowValues);
+            String rowString = getRowString(rowValues);
+            printWriter.println(rowString);
         }
-    }
-
-    public void setResponseHeaders(Response response) {
-        response.type(MEDIA_TYPE);
-        response.header("Access-Control-Expose-Headers", "Content-Disposition");
-        response.header("Content-Disposition", "attachment;filename=" + getExportFilename(fileFormat));
+        printWriter.flush();
```

Comment:
I guess `printWriter` is not closed anywhere.

Reply:
That's correct, I'll add a comment to that effect.  Closing PrintWriter would close the underlying OutpuStream, which would then prevent the rest of the ZipFile from being written